### PR TITLE
Updating Combiners.Top PTransform Documentation

### LINKED ISSUE #19819: Remove deprecated 'compare' argument from combiners.Top in PyDocs
PyDoc: [https://beam.apache.org/releases/pydoc/2.12.0/apache_beam.transforms.combiners.html?#apache_beam.transforms.combiners.Top](https://beam.apache.org/releases/pydoc/2.12.0/apache_beam.transforms.combiners.html?#apache_beam.transforms.combiners.Top)

The combiners.Top PyDoc still shows the `compare` argument as usable. It is deprecated and results in an error in Python 3, but the PyDoc does not reflect that.

It should say that the argument is deprecated in favor of using the `key` and `reverse` arguments.

Imported from Jira [BEAM-8784](https://issues.apache.org/jira/browse/BEAM-8784). Original Jira may contain additional context.
Reported by: dcavazos.

### PR DESCRIPTION
**Pull request by svetakvsundhar**
Fixes documentation in #19819. 

TODO: A follow up PR that removes the compare argument from functions. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by svetakvsundhar** [comment]
r: @tvalentyn

**Comment by svetakvsundhar** [comment]
Run Python_Runners PreCommit

**Comment by tvalentyn** [comment]
per https://github.com/apache/beam/issues/19819#issuecomment-1367446140 we should remove references to `compare` instead of adding new ones.

also, Beam is  Py3 only now.

File: sdks/python/apache_beam/transforms/combiners.py [hunk 1]
```diff
@@ -195,11 +195,12 @@ class Top(object):
   @with_input_types(T)
   @with_output_types(List[T])
   class Of(CombinerWithoutDefaults):
-    """Obtain a list of the compare-most N elements in a PCollection.
+    """Obtain a list of the n greatest elements in a PCollection.
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/combiners.py:198]:
```suggestion
    """Obtain a list of the N greatest elements in a PCollection.
```

File: sdks/python/apache_beam/transforms/combiners.py [hunk 2]
```diff
@@ -195,11 +195,12 @@ class Top(object):
...
 
     This transform will retrieve the n greatest elements in the PCollection
-    to which it is applied, where "greatest" is determined by the comparator
-    function supplied as the compare argument.
+    to which it is applied, where "greatest" is determined by a
+    function supplied as the key or reverse arguments.
+
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/combiners.py:203]: nit: extra newline

File: sdks/python/apache_beam/transforms/combiners.py [hunk 3]
```diff
@@ -249,12 +250,13 @@ def expand(self, pcoll):
...
-    """Identifies the compare-most N elements associated with each key.
+    """Identifies the N greatest elements associated with each key.
 
     This transform will produce a PCollection mapping unique keys in the input
     PCollection to the n greatest elements with which they are associated, where
-    "greatest" is determined by the comparator function supplied as the compare
-    argument in the initializer.
+    "greatest" is determined by a function supplied as the key or
+    reverse arguments.
+
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/combiners.py:259]: nit: extra newline

File: sdks/python/apache_beam/transforms/combiners.py [hunk 4]
```diff
@@ -249,12 +250,13 @@ def expand(self, pcoll):
...
   @with_output_types(Tuple[K, List[V]])
   class PerKey(ptransform.PTransform):
-    """Identifies the compare-most N elements associated with each key.
+    """Identifies the N greatest elements associated with each key.
 
     This transform will produce a PCollection mapping unique keys in the input
     PCollection to the n greatest elements with which they are associated, where
-    "greatest" is determined by the comparator function supplied as the compare
-    argument in the initializer.
+    "greatest" is determined by a function supplied as the key or
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/combiners.py:257]: let's back-tick param names. eg. \`key\`

**Comment by tvalentyn** [comment]
manually checked that docs, lint and formatter precommits passed.

**Pull request merged**